### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Fahrenhei7**
Done:

- [x] Devise installed;
- [x] Simple form installed;
- [x] Created user model;
- [x] Added custom controllers for devise, to add User::name to strong params and add possibility to custom redirects after signing in / signing up / etc;
- [x] Added authentication with User::name attribute;
- [x] Added simple validation to User::name, to avoid conflicts with User::email.
- [x] Convert devise views from `.erb` to `.slim`;
- [x] Add acceptance tests;
- [x] Add some styling to views, add a header to a layout;

To implement signing in with a name I used [this](https://github.com/plataformatec/devise/wiki/How-To:-Allow-users-to-sign-in-using-their-username-or-email-address) article.

Todo:

- [ ] Move default devise views folder to `app/views/web/devise` somehow;
- [ ] Add length validation to User::name;
- [ ] Add something, that I forgot to write here.

### CODE REVIEW

File: app/views/web/static_pages/welcome.slim [hunk 1]
```diff
@@ -0,0 +1,8 @@
+scss:
```
Comment (Freika) [inline app/views/web/static_pages/welcome.slim]: no way using css in views

File: spec/factories/users.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :user do
+
```
Comment (Freika) [inline spec/factories/users.rb]: Fill user factory

File: app/views/web/static_pages/welcome.slim (hunk 1 above)
Comment (Fahrenhei7) [inline app/views/web/static_pages/welcome.slim]: come on. this template would change in next few commits.

File: app/views/devise/mailer/confirmation_instructions.html.slim [hunk 3]
```diff
@@ -0,0 +1,8 @@
+p
+  | Welcome
```
Comment (Freika) [inline app/views/devise/mailer/confirmation_instructions.html.slim]: `"Welcome #{@email}!`

File: app/views/devise/mailer/password_change.html.slim [hunk 4]
```diff
@@ -0,0 +1,6 @@
+p
+  | Hello
```
Comment (Freika) [inline app/views/devise/mailer/password_change.html.slim]: https://github.com/ruruby/onenewthing/pull/1/files#r77458740 same

File: app/views/devise/mailer/reset_password_instructions.html.slim [hunk 5]
```diff
@@ -0,0 +1,12 @@
+p
+  | Hello
```
Comment (Freika) [inline app/views/devise/mailer/reset_password_instructions.html.slim]: https://github.com/ruruby/onenewthing/pull/1/files#r77458740 same

File: app/views/devise/mailer/unlock_instructions.html.slim [hunk 6]
```diff
@@ -0,0 +1,10 @@
+p
+  | Hello
```
Comment (Freika) [inline app/views/devise/mailer/unlock_instructions.html.slim]: https://github.com/ruruby/onenewthing/pull/1/files#r77458740 same

File: spec/rails_helper.rb [hunk 7]
```diff
@@ -35,6 +34,7 @@
       example.run
     end
   end
+
```
Comment (Freika) [inline spec/rails_helper.rb]: Remove empty line

**Pull request merged**